### PR TITLE
interfaces/opengl: add support for ARM Mali

### PR DESCRIPTION
**Pull request by EthanHsieh**
$ udevadm info --query property /dev/mali0
DEVPATH=/devices/platform/soc/13040000.mali/misc/mali0
DEVNAME=/dev/mali0
DEVMODE=0666
MAJOR=10
MINOR=60
SUBSYSTEM=misc
USEC_INITIALIZED=5223664
TAGS=:snap_ubuntu-frame_ubuntu-frame:snap_mir-test-tools_qt-test:snap_mir-test-tools_sdl2-test:snap_mir-test-tools_smoke-test:snap_mir-test-tools_hook_configure:snap_ubuntu-frame_daemon:snap_mir-test-tools_gtk3-test:snap_mir-test-tools_performance-test:

dma_buf_test_exportor is a necessary kernel module since ARM Mali r32p0
It's the buffer allocator and can export a DMA buffer.
/dev/dma_buf_te is generated by dma_buf_test_exportor
$ udevadm info --query property /dev/dma_buf_te
DEVPATH=/devices/virtual/misc/dma_buf_te
DEVNAME=/dev/dma_buf_te
MAJOR=10
MINOR=58
SUBSYSTEM=misc
USEC_INITIALIZED=4964986
TAGS=:snap_ubuntu-frame_ubuntu-frame:snap_mir-test-tools_hook_configure:snap_ubuntu-frame_daemon:snap_mir-test-tools_performance-test:snap_mir-test-tools_qt-test:snap_mir-test-tools_gtk3-test:snap_mir-test-tools_smoke-test:snap_mir-test-tools_sdl2-test:

### CODE REVIEW

File: interfaces/builtin/opengl.go [hunk 1]
```diff
@@ -117,6 +117,10 @@ unix (bind,listen) type=seqpacket addr="@cuda-uvmfd-[0-9a-f]*",
 # Imagination PowerVR driver
 /dev/pvr_sync rw,
 
+# ARM Mali driver
+/dev/mali[0-9]* rw,
+/dev/dma_buf_te rw,
```
Comment (mardy) [inline interfaces/builtin/opengl.go:122]: Please also extend the unit test for checking these (or at least one of them).

Comment (EthanHsieh) [reply interfaces/builtin/opengl.go:122]: @mardy 
Do you mean https://github.com/snapcore/snapd/blob/master/interfaces/builtin/opengl_test.go#L83?
It seems to me that the unit test for mail can be covered by L83 & L84.

Comment (mardy) [reply interfaces/builtin/opengl.go:122]: Yes, I would add at least one line checking for `/dev/mali[0-9]*`

Comment (EthanHsieh) [reply interfaces/builtin/opengl.go:122]: @mardy I have added a commit for it.
Here is the test result.
$ ./run-checks
...
All good, what could possibly go wrong.

btw... for the first commit, I have built a test snapd and verified it on the device with Mali. It worked.
Here is the detailed information: https://bugs.launchpad.net/baoshan/+bug/1958487

Comment (mardy) [reply interfaces/builtin/opengl.go:122]: Thanks!

**Review by mardy — APPROVED** [review]
LGTM, thanks!

**Review by anonymouse64 — APPROVED** [review]
lgtm

**Review by alexmurray — APPROVED** [review]
LGTM!

**Pull request merged**